switchlink/switchlink_link_test.cc

```diff
@@ -36,79 +36,96 @@ extern "C" {
     sequentially, not in parallel, or (2) each test case is run in a
     separate process. If that's not the case, we will need to come up
     with a more creative solution (such as thread-specific storage).
+
+    TODO: Look into using CMock instead of the clumsy methodology used
+    in this test.
 ******************************************************************************/
 
 enum handler_type {
-  CREATE_INTERFACE = 1,
-  DELETE_INTERFACE = 2,
-  CREATE_TUNNEL_INTERFACE = 3,
-  DELETE_TUNNEL_INTERFACE = 4,
+  CREATE_INTERFACE = 1,           // switchlink_create_interface
+  DELETE_INTERFACE = 2,           // switchlink_delete_interface
+  CREATE_TUNNEL_INTERFACE = 3,    // switchlink_create_tunnel_interface
+  DELETE_TUNNEL_INTERFACE = 4,    // switchlink_delete_tunnel_interface
+  CREATE_VRF = 5,                 // switchlink_create_vrf
 };
 
 /**
- * Result variables.
+ * Test results structure. Records the dummy function (handler) called and
+ * its parameter values.
  */
 struct test_results {
-  // Parameter values
-  switchlink_db_tunnel_interface_info_t tunnel_info;
-  switchlink_db_interface_info_t interface_info;
-  uint32_t ifindex;
+  union {
+    // Parameter values
+    switchlink_db_tunnel_interface_info_t tunnel_info;
```

Comment:
nit: organize in dictionary order (aligns with `handler_type` definition order)

Reply:
I don't think so. The correlation between the two lists is too weak and non-obvious to be of any real use.

| handler_type | parameter value |
| ------------ | -------------- |
| CREATE_INTERFACE | switchlink_db_interface_info_t interface_info |
| DELETE_INTERFACE | uint32_t ifindex |
| CREATE_TUNNEL_INTERFACE | switchlink_db_tunnel_interface_info_t tunnel_info |
| DELETE_TUNNEL_INTERFACE | uint32_t ifindex |
| CREATE_VRF | |